### DISCUSSION TITLE
关于按需加载和打包体积的问题（使用create-react-app）

### DISCUSSION THREAD

**Discussion by alwilda** [ant-design/ant-design#29155] [Q&A] [answered]
引入并使用`antd`（只写了一个页面）后启动项目，控制台会有`ou are using a whole package of antd, please use...`的警告，但此时打包后（约500kb），分析如下：
[![yZWz5Q.md.jpg](https://s3.ax1x.com/2021/02/01/yZWz5Q.md.jpg)](https://imgchr.com/i/yZWz5Q)

通过`craco`引入`babel-plugin-import`后，配置如下（craco.config.js）：
```js
module.exports = {
  babel: {
    plugins: [
      ["import", {
        "libraryDirectory": "lib",
        "libraryName": "antd",
        "style": true,
      }]
    ]
  },
}
```
此时再次启动项目后**警告消失**，但是打包后分析如下（约1.5MB）：
[![yZhZOP.md.jpg](https://s3.ax1x.com/2021/02/01/yZhZOP.md.jpg)](https://imgchr.com/i/yZhZOP)

运行时前者也自然比后者加载的快。

这正常吗？还是配置的有问题？亦或是其他原因...

项目版本：
- "antd": "^4.11.2",
-  "@craco/craco": "^6.0.0"
- "babel-plugin-import": "^1.13.3"
- "react": "^17.0.1"

**Comment by xrkffgg** [comment, marked as answer]
https://github.com/ant-design/ant-design/pull/29045
